src/enrichment_tables/pgtable.rs

```diff
@@ -0,0 +1,521 @@
+//! Handles enrichment tables for `type = pgtable`.
+use std::{collections::HashMap, hash::Hasher};
+use std::{path::PathBuf};
+use bytes::Bytes;
+use vector_lib::enrichment::{Case, Condition, IndexHandle, Table};
+use itertools::intersperse;
+
+// use bytes::Bytes;
+// use tracing::trace;
+use vector_lib::configurable::configurable_component;
+// use vector_lib::{conversion::Conversion, TimeZone};
+// use vrl::value::{ObjectMap, Value};
+use chrono::{DateTime, Utc};
+use crate::config::EnrichmentTableConfig;
+use snafu::{Snafu};
+use postgres_openssl::MakeTlsConnector;
+use tokio_postgres::{Config, Error as PgError, NoTls, Row, Statement, Client};
+use tokio_postgres::types::Type;
+use openssl::{
+    error::ErrorStack,
+    ssl::{SslConnector, SslMethod},
+};
+use vrl::value::{ObjectMap, Value};
+use futures::executor::block_on;
+
+#[derive(Debug, Snafu)]
+enum ConnectError {
+    #[snafu(display("failed to create tls connector: {}", source))]
+    TlsFailed { source: ErrorStack },
+    #[snafu(display("failed to connect ({}): {}", endpoint, source))]
+    ConnectionFailed { source: PgError, endpoint: String },
+}
+
+#[derive(Debug, Snafu)]
+enum BuildError {
+    #[snafu(display("invalid endpoint: {}", source))]
+    InvalidEndpoint { source: PgError },
+    // #[snafu(display("host missing"))]
+    // HostMissing,
+    // #[snafu(display("multiple hosts not supported: {:?}", hosts))]
+    // MultipleHostsNotSupported { hosts: Vec<Host> },
+}
+
+/// Configuration of TLS when connecting to PostgreSQL.
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+struct PgtableTlsConfig {
+    /// Absolute path to an additional CA certificate file.
+    ///
+    /// The certificate must be in the DER or PEM (X.509) format.
+    #[configurable(metadata(docs::examples = "certs/ca.pem"))]
+    ca_file: PathBuf,
+}
+
+/// Postgres-specific settings
+#[configurable_component(enrichment_table("pgtable"))]
+#[derive(Clone, Debug, Default)]
+pub struct PgtableConfig {
```

Comment:
I think it would be interesting that the user can specify a custom query and not just columns to select (I'm thinking in aggregations, filtering and so...). This will allow more flexibility, but I may be wrong

@StephenWakely ?

Reply:
Yes. Could definitely be useful.